### PR TITLE
Find or create by (and find or initialize by)

### PR DESCRIPTION
**Pull request by AndyRosenberg**

### CODE REVIEW

File: spec/granite/integrators/find_or_spec.cr [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe "find_or_create_by, find_or_initialize_by" do
+  it "creates on find_or_create when not found" do
+    Parent.clear
+    Parent.find_or_create_by(name: "name")
+    Parent.all.first.name.should eq("name")
+    Parent.all.first.new_record?.should eq(false)
+  end
+
+  it "uses find on find_or_create_by when it exists" do
+    Parent.find_or_create_by(name: "name")
```
Comment (drujensen) [inline spec/granite/integrators/find_or_spec.cr]: Each test should be able to be ran by itself.  I recommend performing a `clear`, `create` and then `find_or_create_by` in this test.

File: spec/granite/integrators/find_or_spec.cr [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require "../../spec_helper"
+
+describe "find_or_create_by, find_or_initialize_by" do
+  it "creates on find_or_create when not found" do
+    Parent.clear
+    Parent.find_or_create_by(name: "name")
+    Parent.all.first.name.should eq("name")
+    Parent.all.first.new_record?.should eq(false)
```
Comment (Blacksmoke16) [inline spec/granite/integrators/find_or_spec.cr]: Super minor but `Parent.all.first` could be shortened a bit to `Parent.first!`.

File: spec/granite/integrators/find_or_spec.cr (hunk 1 above)
Comment (AndyRosenberg) [reply spec/granite/integrators/find_or_spec.cr]: Done with before_each block.

File: spec/granite/integrators/find_or_spec.cr (hunk 2 above)
Comment (AndyRosenberg) [reply spec/granite/integrators/find_or_spec.cr]: fixed in latest

**Comment by AndyRosenberg** [comment]
Travis is failing with "Closed stream (IO::Error)", not happening for me locally.

File: spec/granite/integrators/find_or_spec.cr [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require "../../spec_helper"
+
+describe "find_or_create_by, find_or_initialize_by" do
+  Spec.before_each do
+    Parent.clear
```
Comment (drujensen) [inline spec/granite/integrators/find_or_spec.cr]: try removing the `before_each` and putting the `Parent.clear` in each test.  I'm interested to see if that fixes it.

Comment (robacarp) [reply spec/granite/integrators/find_or_spec.cr]: Remember`Spec.before_each` isn't scoped to the describe block, it runs before each test _application wide_.

**Pull request merged**